### PR TITLE
Fully upgrade to 1.0

### LINKED ISSUE #204: common.jl fails to run
Now that [issue #202](https://github.com/ihnorton/Clang.jl/issues/202) is fixed, the file **common.jl** gets autogenerated.
But *common.jl* has the following problem:
```
$ julia common.jl 
ERROR: LoadError: InexactError: trunc(Int32, 9223372036854775807)
Stacktrace:
 [1] top-level scope at none:0
in expression starting at /home/jose/src/julia/vapoursynth/common.jl:44
```

The relevant lines are (lines 41-45):
```julia
# Skipping MacroDefinition: UINT_LEAST64_MAX ( __UINT64_C ( 18446744073709551615 ) )

const INT_FAST8_MIN = -128
const INT_FAST16_MIN = -(Int32(9223372036854775807)) - 1
const INT_FAST32_MIN = -(Int32(9223372036854775807)) - 1
```
In fact, if comment line 44, the problem appears in more places. 

The full **common.jl** is:
```julia
# Automatically generated using Clang.jl wrap_c, version 0.0.0

using Compat

const unix = 1
const linux = 1
const INT8_MIN = -128
const INT16_MIN = -32767 - 1
const INT32_MIN = -2147483647 - 1

# Skipping MacroDefinition: INT64_MIN ( - __INT64_C ( 9223372036854775807 ) - 1 )

const INT8_MAX = 127
const INT16_MAX = 32767
const INT32_MAX = 2147483647

# Skipping MacroDefinition: INT64_MAX ( __INT64_C ( 9223372036854775807 ) )

const UINT8_MAX = 255
const UINT16_MAX = 65535
const UINT32_MAX = UInt32(4294967295)

# Skipping MacroDefinition: UINT64_MAX ( __UINT64_C ( 18446744073709551615 ) )

const INT_LEAST8_MIN = -128
const INT_LEAST16_MIN = -32767 - 1
const INT_LEAST32_MIN = -2147483647 - 1

# Skipping MacroDefinition: INT_LEAST64_MIN ( - __INT64_C ( 9223372036854775807 ) - 1 )

const INT_LEAST8_MAX = 127
const INT_LEAST16_MAX = 32767
const INT_LEAST32_MAX = 2147483647

# Skipping MacroDefinition: INT_LEAST64_MAX ( __INT64_C ( 9223372036854775807 ) )

const UINT_LEAST8_MAX = 255
const UINT_LEAST16_MAX = 65535
const UINT_LEAST32_MAX = UInt32(4294967295)

# Skipping MacroDefinition: UINT_LEAST64_MAX ( __UINT64_C ( 18446744073709551615 ) )

const INT_FAST8_MIN = -128
const INT_FAST16_MIN = -(Int32(9223372036854775807)) - 1
const INT_FAST32_MIN = -(Int32(9223372036854775807)) - 1

# Skipping MacroDefinition: INT_FAST64_MIN ( - __INT64_C ( 9223372036854775807 ) - 1 )

const INT_FAST8_MAX = 127
const INT_FAST16_MAX = Int32(9223372036854775807)
const INT_FAST32_MAX = Int32(9223372036854775807)

# Skipping MacroDefinition: INT_FAST64_MAX ( __INT64_C ( 9223372036854775807 ) )

const UINT_FAST8_MAX = 255
const UINT_FAST16_MAX = UInt32(@int128_str("18446744073709551615"))
const UINT_FAST32_MAX = UInt32(@int128_str("18446744073709551615"))

# Skipping MacroDefinition: UINT_FAST64_MAX ( __UINT64_C ( 18446744073709551615 ) )

const INTPTR_MIN = -(Int32(9223372036854775807)) - 1
const INTPTR_MAX = Int32(9223372036854775807)
const UINTPTR_MAX = UInt32(@int128_str("18446744073709551615"))

# Skipping MacroDefinition: INTMAX_MIN ( - __INT64_C ( 9223372036854775807 ) - 1 )
# Skipping MacroDefinition: INTMAX_MAX ( __INT64_C ( 9223372036854775807 ) )
# Skipping MacroDefinition: UINTMAX_MAX ( __UINT64_C ( 18446744073709551615 ) )

const PTRDIFF_MIN = -(Int32(9223372036854775807)) - 1
const PTRDIFF_MAX = Int32(9223372036854775807)
const SIG_ATOMIC_MIN = -2147483647 - 1
const SIG_ATOMIC_MAX = 2147483647
const SIZE_MAX = UInt32(@int128_str("18446744073709551615"))
const WCHAR_MIN = __WCHAR_MIN
const WCHAR_MAX = __WCHAR_MAX
const WINT_MIN = UInt32(0)
const WINT_MAX = UInt32(4294967295)

# Skipping MacroDefinition: INT8_C ( c ) c
# Skipping MacroDefinition: INT16_C ( c ) c
# Skipping MacroDefinition: INT32_C ( c ) c
# Skipping MacroDefinition: INT64_C ( c ) c ## L
# Skipping MacroDefinition: UINT8_C ( c ) c
# Skipping MacroDefinition: UINT16_C ( c ) c
# Skipping MacroDefinition: UINT32_C ( c ) c ## U
# Skipping MacroDefinition: UINT64_C ( c ) c ## UL
# Skipping MacroDefinition: INTMAX_C ( c ) c ## L
# Skipping MacroDefinition: UINTMAX_C ( c ) c ## UL

const VAPOURSYNTH_API_MAJOR = 3
const VAPOURSYNTH_API_MINOR = 5

# Skipping MacroDefinition: VAPOURSYNTH_API_VERSION ( ( VAPOURSYNTH_API_MAJOR << 16 ) | ( VAPOURSYNTH_API_MINOR ) )
# Skipping MacroDefinition: VS_EXTERNAL_API ( ret ) VS_EXTERN_C __attribute__ ( ( visibility ( "default" ) ) ) ret VS_CC
# Skipping MacroDefinition: VS_API ( ret ) VS_EXTERNAL_API ( ret )

const int8_t = __int8_t
const int16_t = __int16_t
const int32_t = __int32_t
const int64_t = __int64_t
const uint8_t = __uint8_t
const uint16_t = __uint16_t
const uint32_t = __uint32_t
const uint64_t = __uint64_t
const int_least8_t = __int_least8_t
const int_least16_t = __int_least16_t
const int_least32_t = __int_least32_t
const int_least64_t = __int_least64_t
const uint_least8_t = __uint_least8_t
const uint_least16_t = __uint_least16_t
const uint_least32_t = __uint_least32_t
const uint_least64_t = __uint_least64_t
const int_fast8_t = UInt8
const int_fast16_t = Clong
const int_fast32_t = Clong
const int_fast64_t = Clong
const uint_fast8_t = Cuchar
const uint_fast16_t = Culong
const uint_fast32_t = Culong
const uint_fast64_t = Culong
const intptr_t = Clong
const uintptr_t = Culong
const intmax_t = __intmax_t
const uintmax_t = __uintmax_t

$(Expr(:type, true, :VSFrameRef, quote
end))

$(Expr(:type, true, :VSNodeRef, quote
end))

$(Expr(:type, true, :VSCore, quote
end))

$(Expr(:type, true, :VSPlugin, quote
end))

$(Expr(:type, true, :VSNode, quote
end))

$(Expr(:type, true, :VSFuncRef, quote
end))

$(Expr(:type, true, :VSMap, quote
end))

$(Expr(:type, true, :VSAPI, quote
    createCore::Ptr{Cvoid}
    freeCore::Ptr{Cvoid}
    getCoreInfo::Ptr{Cvoid}
    cloneFrameRef::Ptr{Cvoid}
    cloneNodeRef::Ptr{Cvoid}
    cloneFuncRef::Ptr{Cvoid}
    freeFrame::Ptr{Cvoid}
    freeNode::Ptr{Cvoid}
    freeFunc::Ptr{Cvoid}
    newVideoFrame::Ptr{Cvoid}
    copyFrame::Ptr{Cvoid}
    copyFrameProps::Ptr{Cvoid}
    registerFunction::Ptr{Cvoid}
    getPluginById::Ptr{Cvoid}
    getPluginByNs::Ptr{Cvoid}
    getPlugins::Ptr{Cvoid}
    getFunctions::Ptr{Cvoid}
    createFilter::Ptr{Cvoid}
    setError::Ptr{Cvoid}
    getError::Ptr{Cvoid}
    setFilterError::Ptr{Cvoid}
    invoke::Ptr{Cvoid}
    getFormatPreset::Ptr{Cvoid}
    registerFormat::Ptr{Cvoid}
    getFrame::Ptr{Cvoid}
    getFrameAsync::Ptr{Cvoid}
    getFrameFilter::Ptr{Cvoid}
    requestFrameFilter::Ptr{Cvoid}
    queryCompletedFrame::Ptr{Cvoid}
    releaseFrameEarly::Ptr{Cvoid}
    getStride::Ptr{Cvoid}
    getReadPtr::Ptr{Cvoid}
    getWritePtr::Ptr{Cvoid}
    createFunc::Ptr{Cvoid}
    callFunc::Ptr{Cvoid}
    createMap::Ptr{Cvoid}
    freeMap::Ptr{Cvoid}
    clearMap::Ptr{Cvoid}
    getVideoInfo::Ptr{Cvoid}
    setVideoInfo::Ptr{Cvoid}
    getFrameFormat::Ptr{Cvoid}
    getFrameWidth::Ptr{Cvoid}
    getFrameHeight::Ptr{Cvoid}
    getFramePropsRO::Ptr{Cvoid}
    getFramePropsRW::Ptr{Cvoid}
    propNumKeys::Ptr{Cvoid}
    propGetKey::Ptr{Cvoid}
    propNumElements::Ptr{Cvoid}
    propGetType::Ptr{Cvoid}
    propGetInt::Ptr{Cvoid}
    propGetFloat::Ptr{Cvoid}
    propGetData::Ptr{Cvoid}
    propGetDataSize::Ptr{Cvoid}
    propGetNode::Ptr{Cvoid}
    propGetFrame::Ptr{Cvoid}
    propGetFunc::Ptr{Cvoid}
    propDeleteKey::Ptr{Cvoid}
    propSetInt::Ptr{Cvoid}
    propSetFloat::Ptr{Cvoid}
    propSetData::Ptr{Cvoid}
    propSetNode::Ptr{Cvoid}
    propSetFrame::Ptr{Cvoid}
    propSetFunc::Ptr{Cvoid}
    setMaxCacheSize::Ptr{Cvoid}
    getOutputIndex::Ptr{Cvoid}
    newVideoFrame2::Ptr{Cvoid}
    setMessageHandler::Ptr{Cvoid}
    setThreadCount::Ptr{Cvoid}
    getPluginPath::Ptr{Cvoid}
    propGetIntArray::Ptr{Cvoid}
    propGetFloatArray::Ptr{Cvoid}
    propSetIntArray::Ptr{Cvoid}
    propSetFloatArray::Ptr{Cvoid}
    logMessage::Ptr{Cvoid}
end))

$(Expr(:type, true, :VSFrameContext, quote
end))

# begin enum VSColorFamily
const VSColorFamily = UInt32
const cmGray = (UInt32)(1000000)
const cmRGB = (UInt32)(2000000)
const cmYUV = (UInt32)(3000000)
const cmYCoCg = (UInt32)(4000000)
const cmCompat = (UInt32)(9000000)
# end enum VSColorFamily

# begin enum VSSampleType
const VSSampleType = UInt32
const stInteger = (UInt32)(0)
const stFloat = (UInt32)(1)
# end enum VSSampleType

# begin enum VSPresetFormat
const VSPresetFormat = UInt32
const pfNone = (UInt32)(0)
const pfGray8 = (UInt32)(1000010)
const pfGray16 = (UInt32)(1000011)
const pfGrayH = (UInt32)(1000012)
const pfGrayS = (UInt32)(1000013)
const pfYUV420P8 = (UInt32)(3000010)
const pfYUV422P8 = (UInt32)(3000011)
const pfYUV444P8 = (UInt32)(3000012)
const pfYUV410P8 = (UInt32)(3000013)
const pfYUV411P8 = (UInt32)(3000014)
const pfYUV440P8 = (UInt32)(3000015)
const pfYUV420P9 = (UInt32)(3000016)
const pfYUV422P9 = (UInt32)(3000017)
const pfYUV444P9 = (UInt32)(3000018)
const pfYUV420P10 = (UInt32)(3000019)
const pfYUV422P10 = (UInt32)(3000020)
const pfYUV444P10 = (UInt32)(3000021)
const pfYUV420P16 = (UInt32)(3000022)
const pfYUV422P16 = (UInt32)(3000023)
const pfYUV444P16 = (UInt32)(3000024)
const pfYUV444PH = (UInt32)(3000025)
const pfYUV444PS = (UInt32)(3000026)
const pfYUV420P12 = (UInt32)(3000027)
const pfYUV422P12 = (UInt32)(3000028)
const pfYUV444P12 = (UInt32)(3000029)
const pfYUV420P14 = (UInt32)(3000030)
const pfYUV422P14 = (UInt32)(3000031)
const pfYUV444P14 = (UInt32)(3000032)
const pfRGB24 = (UInt32)(2000010)
const pfRGB27 = (UInt32)(2000011)
const pfRGB30 = (UInt32)(2000012)
const pfRGB48 = (UInt32)(2000013)
const pfRGBH = (UInt32)(2000014)
const pfRGBS = (UInt32)(2000015)
const pfCompatBGR32 = (UInt32)(9000010)
const pfCompatYUY2 = (UInt32)(9000011)
# end enum VSPresetFormat

# begin enum VSFilterMode
const VSFilterMode = UInt32
const fmParallel = (UInt32)(100)
const fmParallelRequests = (UInt32)(200)
const fmUnordered = (UInt32)(300)
const fmSerial = (UInt32)(400)
# end enum VSFilterMode

$(Expr(:type, true, :VSFormat, quote
    name::NTuple{32, UInt8}
    id::Cint
    colorFamily::Cint
    sampleType::Cint
    bitsPerSample::Cint
    bytesPerSample::Cint
    subSamplingW::Cint
    subSamplingH::Cint
    numPlanes::Cint
end))

# begin enum VSNodeFlags
const VSNodeFlags = UInt32
const nfNoCache = (UInt32)(1)
const nfIsCache = (UInt32)(2)
const nfMakeLinear = (UInt32)(4)
# end enum VSNodeFlags

# begin enum VSPropTypes
const VSPropTypes = UInt32
const ptUnset = (UInt32)(117)
const ptInt = (UInt32)(105)
const ptFloat = (UInt32)(102)
const ptData = (UInt32)(115)
const ptNode = (UInt32)(99)
const ptFrame = (UInt32)(118)
const ptFunction = (UInt32)(109)
# end enum VSPropTypes

# begin enum VSGetPropErrors
const VSGetPropErrors = UInt32
const peUnset = (UInt32)(1)
const peType = (UInt32)(2)
const peIndex = (UInt32)(4)
# end enum VSGetPropErrors

# begin enum VSPropAppendMode
const VSPropAppendMode = UInt32
const paReplace = (UInt32)(0)
const paAppend = (UInt32)(1)
const paTouch = (UInt32)(2)
# end enum VSPropAppendMode

$(Expr(:type, true, :VSCoreInfo, quote
    versionString::Cstring
    core::Cint
    api::Cint
    numThreads::Cint
    maxFramebufferSize::Int64
    usedFramebufferSize::Int64
end))

$(Expr(:type, true, :VSVideoInfo, quote
    format::Ptr{VSFormat}
    fpsNum::Int64
    fpsDen::Int64
    width::Cint
    height::Cint
    numFrames::Cint
    flags::Cint
end))

# begin enum VSActivationReason
const VSActivationReason = Cint
const arInitial = (Int32)(0)
const arFrameReady = (Int32)(1)
const arAllFramesReady = (Int32)(2)
const arError = (Int32)(-1)
# end enum VSActivationReason

# begin enum VSMessageType
const VSMessageType = UInt32
const mtDebug = (UInt32)(0)
const mtWarning = (UInt32)(1)
const mtCritical = (UInt32)(2)
const mtFatal = (UInt32)(3)
# end enum VSMessageType

const VSGetVapourSynthAPI = Ptr{Cvoid}
const VSPublicFunction = Ptr{Cvoid}
const VSRegisterFunction = Ptr{Cvoid}
const VSConfigPlugin = Ptr{Cvoid}
const VSInitPlugin = Ptr{Cvoid}
const VSFreeFuncData = Ptr{Cvoid}
const VSFilterInit = Ptr{Cvoid}
const VSFilterGetFrame = Ptr{Cvoid}
const VSFilterFree = Ptr{Cvoid}
const VSFrameDoneCallback = Ptr{Cvoid}
const VSMessageHandler = Ptr{Cvoid}
```

### PR DESCRIPTION
**Pull request by Gnimuc**
This branch should fix #204 for now, I gonna check other parts later on. Another concern is the enum integration(#162),  currently, it seems that it's on nobody's radar.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,9 +20,9 @@ matrix:
 addons:
   apt:
     packages:
-      llvm-3.9
-      llvm-3.9-tools
-      libclang-3.9-dev
+      llvm-6.0
```
Comment (ihnorton) [inline .travis.yml]: Looks like these aren't available in the travis image.

**Comment by ihnorton** [comment]
Thanks. I took a quick glance, 😐on the style changes (but that's fine). LGTM, so feel free to merge when it passes CI.

**Comment by ihnorton** [comment]
(haven't had a lot of time for this recently, so I really appreciate all the fixes -- you should have an invite for push access now)

File: .travis.yml (hunk 1 above)
Comment (Gnimuc) [reply .travis.yml]: The ultimate solution would be BinaryBuilder. I tried LLVMBuilder yesterday, but also hit     https://github.com/staticfloat/LLVMBuilder/issues/33 . Then, I tried LLVM official binaries and it worked well. However, LLVM doesn't officially ship generic Linux binaries, so we still need to find a way to work around LLVMBuilder#33.

**Comment by Gnimuc** [comment]
hi @ihnorton I can only `Squash and merge` this PR instead of a common `merge commit`, but I guess those commits should be reserved since there are so many changes in this PR. Could you help to merge this?

**Comment by ihnorton** [comment]
Thanks! (`Rebase and merge` should have been available too, I think)

**Pull request merged**